homeassistant/components/livisi/switch.py

```diff
@@ -0,0 +1,185 @@
+
+    async def async_discover_device(device) -> None:
+        """Add switch."""
+        livisi_switch: SwitchEntity = await create_entity(
+            hass, config_entry, device, coordinator
+        )
+        LOGGER.debug("Include device type: %s", device.get("type"))
+        async_add_entities([livisi_switch])
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(hass, LIVISI_DISCOVERY_NEW, async_discover_device)
+    )
+
+
+async def create_entity(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    device: dict[str, Any],
+    coordinator: LivisiDataUpdateCoordinator,
+) -> SwitchEntity:
+    """Create Switch Entity."""
+    config_details: dict[str, Any] = device["config"]
+    capabilities: list = device["capabilities"]
+    is_on = await coordinator.async_get_pss_state(capability=capabilities[0])
+    room_id: str = device["location"].removeprefix("/location/")
+    room_name: str = coordinator.rooms[room_id]
+    livisi_switch = LivisiSwitch(
+        hass,
+        config_entry,
+        coordinator,
+        unique_id=device.get("id"),
+        manufacturer=device.get("manufacturer"),
+        product=device.get("product"),
+        serial_number=device.get("serialNumber"),
+        device_type=device.get("type"),
+        name=config_details.get("name"),
+        capability_id=capabilities[0],
+        is_on=is_on,
+        room=room_name,
+    )
+    return livisi_switch
+
+
+class LivisiSwitch(SwitchEntity, CoordinatorEntity):
+    """Represents the Livisi Switch."""
+
+    def __init__(
+        self,
+        hass,
+        config_entry: ConfigEntry,
+        coordinator: LivisiDataUpdateCoordinator,
+        unique_id,
+        manufacturer,
+        product,
+        serial_number,
+        device_type,
+        name,
+        capability_id,
+        is_on,
+        room,
+        version=None,
+    ):
+        """Initialize the Livisi Switch."""
+        self.hass: HomeAssistant = hass
+        self.config_entry: ConfigEntry = config_entry
+        self._attr_unique_id = unique_id
+        self._manufacturer = manufacturer
+        self._product = product
+        self._serial_number = serial_number
+        self._attr_model = device_type
+        self._attr_name = name
+        self._state = None
+        self._capability_id = capability_id
+        self._attr_is_on = is_on
+        self._room = room
+        self._version = version
+        self.aio_livisi = coordinator.aiolivisi
+        if is_on is None:
+            self._attr_is_available = False
+        else:
+            self._attr_is_available = True
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, str(self._attr_unique_id))},
+            manufacturer=self._manufacturer,
+            model=self._attr_model,
+            name=self._attr_name,
+            suggested_area=self._room,
+            sw_version=self._version,
+            via_device=(DOMAIN, str(self._attr_unique_id)),
```

Comment:
Why are we using the same identifiers in `via_device` and in `identifiers`? If we're using `via_device` that device should be another device than this device.

Reply:
I removed the `via_device` identifier.